### ISSUE TITLE
loss function graph on Colab

### ISSUE THREAD

**Issue by wisdomk** [AlexeyAB/darknet#3216]
I am trying to train the model on Google Colab but I am just struggling with the fact how to show loss function graph on the browser!
yes I used the option `-mjpeg_port 8090 -map` but still couldn't figure out a way to fetch that port!

I tried to use `!curl ipecho.net/plain` and 
`!pip install netifaces
import netifaces as ni
ni.ifaddresses('eth0')
ip = ni.ifaddresses('eth0')[ni.AF_INET][0]['addr']
print ("IP Address>>>>",ip)`

in order to get the remote server ip and tried to use them but with no response at all!

Could someone please mention if he/she got it successfully to show the graph on Colab, or maybe mention the way I should follow?

**Comment by enesyuceyurt** [comment]
You can just see it on the left side of the screen in the files tab. You can open just by clicking chart.png. Or you can open a second notebook in the same device(just choose GPU) and run this code to automatically backup chart.png every 750 seconds to your drive. You should first mount your drive of course. I always use this method so that I can even see the loss chart from my phone by simply opening my google drive.

```
import time
%cd /content/darknet
while (True):
  !cp chart.png /content/drive/'My Drive'
  time.sleep(750)
```

**Comment by RamaRam99** [comment]
> You can just see it on the left side of the screen in the files tab. You can open just by clicking chart.png. Or you can open a second notebook in the same device(just choose GPU) and run this code to automatically backup chart.png every 750 seconds to your drive. You should first mount your drive of course. I always use this method so that I can even see the loss chart from my phone by simply opening my google drive.
> 
> ```
> import time
> %cd /content/darknet
> while (True):
>   !cp chart.png /content/drive/'My Drive'
>   time.sleep(750)
> ```

How would that even work? isn't the second notebook a new separate google colab notebook without darknet in its directory. Even if we downloaded darknet on the new notebook, it doesnt connect it to content/darknet on the first notebook. Am I missing something here?

**Comment by enesyuceyurt** [comment]
I dont know how it works right now, but back then if you opened a second notebook and choose Gpu, it would run on the same virtual machine so you could see whats happening in that virtual machine. You didnt have to download darknet because if its already downloaded in the first notebook. The important question is if they are running on the same virtual machine. If they are, then what i wrote still applies.
